nix/file_handling.md

```diff
@@ -0,0 +1,192 @@
+terminal-friendly editor. In this workshop we are going to use `nano`. It's
+simple and easy to use.
+
+Make sure you're in your home directory (use `cd` and `pwd` to confirm) then type
+
+```console
+$ nano
+```
+
+This is a no-frills editor. Type something! How about a TODO list?
+
+```console
+TODO
+* [x] Learn how to navigate using the terminal
+* [ ] Learn how to create files
+* [ ] Learn about pipes and redirects
+```
+
+At the bottom you'll notice a bunch of different options but we are concerned
+with only two of them: `Write Out` (save) and `Exit`.
+
+The caret (`^`) means the Control key. To save the TODO list, hit `Ctrl+o`, type
+in a name (how about "TODO") and then hit `Enter`. `nano` will report that it
+wrote some number of lines.
+
+Now exit `nano` by typing `Ctrl+x`.
+
+Use `ls` to see what happened:
+
+```console
+$ ls -F
+```
+```console
+Desktop/  Documents/  Downloads/  Music/  Pictures/  Public/  TODO  Templates/  Videos/
+```
+
+There's the TODO list! If you want to edit the todo list, you can open it up in
+`nano` (you can use tab completion for the filename, too!)
+
+```console
+$ nano TODO
+```
+
+Check off the second item on the todo list and then save and exit `nano`. Notice
+that when you hit `Ctrl+o` to save an existing file, `nano` will automatically
+fill in the name of the existing file. If you wanted to "Save As...", you can
+simply change the name in the `Write Out` bar.
+
+## Create an empty file
+
+There are a few ways to create files on the command line. If you want to create
+an "empty" file, you can use `touch`. Try it!
+
+```console
+$ touch newfile
+```
+```console
+Desktop/    Downloads/  Pictures/  TODO        Videos/
+Documents/  Music/      Public/    Templates/  newfile
+```
+
+You can open `newfile` in `nano` to confirm that it's empty. Then just exit out
+using `Ctrl+x` since there's nothing to save!
+
+## Create a directory
+
+To create a new directory, use the `mkdir` command. We can create a `Research/`
+folder in the home directory.
+
+```console
+$ mkdir Research
+```
+```console
+$ ls -F
+```
+```console
+Desktop/    Downloads/  Pictures/  Research/  Templates/  newfile
+Documents/  Music/      Public/    TODO       Videos/
+```
+
+## Remove a file
+
+We don't need that empty file sitting around, we can remove it. To remove a
+file, use the `rm` command:
+
+```console
+$ rm newfile
+```
+
+Did anything happen?
+
+```console
+$ ls -F
+```
+```console
+Desktop/    Downloads/  Pictures/  Research/  Templates/
+Documents/  Music/      Public/    TODO       Videos/
+```
+
+Yes, `newfile` is gone. And this is something to be aware of: there is no
+"Recycle Bin". There is no "Undo". That file is gone.
+
+## Remove a directory
+
+Let's try to remove the `Research` directory we created earlier.
+
+```console
+$ rm Research
+```
+```console
+rm: cannot remove 'Research': Is a directory
+```
+
+`rm` only works with files by default. If you want to remove the directory you
```

Comment:
These are just suggestions: 
1. What about `rmdir`, I know this only works when the directory is empty but in this case it is. Not sure though if this adds something.  
2. What about `-i ` that will basically ask you if you want to remove before removing. (not sure but we can recommend them to ad that on the ~/.bash_rc, like creating an alias alias rm='rm -i' )